gangplank/ocp/cosa-podman.go

```diff
@@ -145,34 +151,70 @@ func podmanRunner(term termChan, cp CosaPodder, envVars []v1.EnvVar) error {
 		},
 	}
 
-	// Ensure that /srv in the COSA container is defined.
-	srvDir := clusterCtx.podmanSrvDir
-	if srvDir == "" {
-		// ioutil.TempDir does not create the directory with the appropriate perms
-		tmpSrvDir := filepath.Join(cosaSrvDir, s.Name)
-		if err := os.MkdirAll(tmpSrvDir, 0777); err != nil {
-			return fmt.Errorf("failed to create emphemeral srv dir for pod: %w", err)
+	// cleanerFn is a function that cleans up any ephemeral directories.
+	type cleanerFn func()
+	// set the default cleanerFn to be a noop.
+	var localCleanFn cleanerFn = func() {}
+
+	if podmanRemote {
+		// Inform COSA that /srv is not an overlay
+		s.Env["COSA_SKIP_OVERLAY"] = "1"
+		// With out an ephemeral overlay, we need to force unprivileged mode to prevent
+		// "Setting xattrs: fsetxattr(security.selinux): Operation not supported" errors.
+		s.Env["FORCE_UNPRIVILEGED"] = "1"
+		l.Info("Remote container will be purely ephemeral")
```

Comment:
What's the motivation for special casing this?

What I did notice is that if we don't mount in a directory from the host (even if it's just a throwaway directory) then the I/O performance is **really** bad. I don't know if it's totally the I/O being that bad or if it's because we trigger some code in `rpm-ostree` that handles networked filesystems (does a check for NFS or FUSE) and behaves differently as a result.



https://github.com/coreos/rpm-ostree/blob/f2b1f2038bade4e44051522ef9788f22cc241a97/src/app/rpmostree-compose-builtin-tree.cxx#L1089-L1094

https://github.com/coreos/rpm-ostree/blob/f2b1f2038bade4e44051522ef9788f22cc241a97/src/app/rpmostree-compose-builtin-tree.cxx#L546-L564

Reply:
Thanks for unintentionally showing a major flaw in the podman handling with this hack:
https://github.com/coreos/coreos-assembler/blob/main/gangplank/ocp/cosa-podman.go#L150-L165

I've changed this now to use the podman API, which worse in the remote scenerio and _much_ cleaner.